### PR TITLE
script + issue +doc

### PR DESCRIPTION
**Pull request by Slowblitz**
fixing #29 and #27 + adding some documentation , and PEP8 ready

### CODE REVIEW

File: ando/checker.py [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
+# File: checker.py
+# Project: ando
+# File Created: Tuesday, 30th June 2020 10:50:05 am
+# Author: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Last Modified: Thursday, 2nd July 2020 1:29:58 pm
+# Modified By: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Copyright - 2020 MIT, Institue de neurosciences de la Timone
+
```
Comment (SylvainTakerkart) [inline ando/checker.py:10]: it seems you've added this everywhere? It looks like it's automatic... Is this intended?
Anyhow, I have three remarks:
- there's a typo at InstituE, and add a capital letter at Neurosciences
- you should add CNRS, next to the copyright line
- about the MIT licence... we should probably discuss this (but I never know which license to choose)

File: ando/engine.py [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+# File: engine.py
+# Project: ando
+# File Created: Tuesday, 30th June 2020 10:50:05 am
+# Author: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Last Modified: Thursday, 2nd July 2020 1:32:33 pm
+# Modified By: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Copyright - 2020 MIT, Institue de neurosciences de la Timone
+
```
Comment (SylvainTakerkart) [inline ando/engine.py:10]: same as with the previous file

File: ando/engine.py [hunk 3]
```diff
@@ -160,11 +189,17 @@ def is_AnDO_R(subpath, level, validate):
...
     Main method of the validator. uses other class methods for checking
     different aspects of the directory path.
 
-    :param names:
+
+    Args:
+        directory ([str]): [names of the directory to check]
+
+    Returns:
+        [bool]: [does the directory adheres to the ando specification]
```
Comment (SylvainTakerkart) [inline ando/engine.py:202]: remove the final s in "adhereS"

File: ando/engine.py [hunk 4]
```diff
@@ -180,15 +215,20 @@ def is_AnDO_verbose(directory):
     """
     Call the function is_AnDO_verbose_Format on every path in the list
 
-    :param names:
+
+    Args:
+        directory ([str]): [names of the directory to check]
+
+    Returns:
+        [bool]: [does the directory adheres to the ando specification]
```
Comment (SylvainTakerkart) [inline ando/engine.py:223]: same as above: adhere, not adheres

File: ando/engine.py [hunk 5]
```diff
@@ -198,10 +238,26 @@ def is_AnDO_verbose_Format(names):
...
+        SubjectError: raised if it doesn't the subject rules
+        SourceError: raised if it doesn't the source rules
+        RawDataError: raised if it doesn't the rawdata rules
+        DerivativeDataError: raised if it doesn't the derivatives rules
+        MetaDataError: raised if it doesn't the metadata rules
+        SourceNotFound: raised if it doesn't the source rules
+
+    Returns:
+        [bool]: true if error is found else false
+        [out]: feedback for the web api
```
Comment (SylvainTakerkart) [inline ando/engine.py:256]: not API!

File: ando/engine.py [hunk 6]
```diff
@@ -198,10 +238,26 @@ def is_AnDO_verbose_Format(names):
     Main method of the validator. uses other class methods for checking
     different aspects of the directory path.
 
-    :param names: list of names founds in the path
+    Args:
+        names ([list]): [names to check]
+
+    Raises:
+        ExperimentError: raised if it doesn't the experiment rules
```
Comment (SylvainTakerkart) [inline ando/engine.py:245]: replace "doesn't" by "does not", here and everywhere...

("doesn't" is only oral langage; in written english, we should use "does not")

Comment (SylvainTakerkart) [reply ando/engine.py:245]: also, the verb is missing in this sentence... "does not RESPECT the experiment rule"... correct everywhere also ;)

File: ando/error.py [hunk 7]
```diff
@@ -1,24 +1,75 @@
...
+# File: error.py
+# Project: ando
+# File Created: Tuesday, 30th June 2020 10:50:05 am
+# Author: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Last Modified: Thursday, 2nd July 2020 1:31:40 pm
+# Modified By: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Copyright - 2020 MIT, Institue de neurosciences de la Timone
+
```
Comment (SylvainTakerkart) [inline ando/error.py:10]: same as above

File: ando/error.py [hunk 8]
```diff
@@ -1,24 +1,75 @@
...
+
+# !/usr/bin/python
 # -*- coding: utf-8 -*-
 
+"""[Exception raise when the name doesn't follow the AnDO specification
```
Comment (SylvainTakerkart) [inline ando/error.py:15]: correct "raise" into "raised" + correct doesn't as above

File: ando/error.py [hunk 9]
```diff
@@ -1,24 +1,75 @@
...
+    self.strout : corresponding to the output of web mod by returning
+     html
+
+    ]
+"""
+
+# flake8: noqa: E501
 class ExperimentError(Exception):
+    """
+    Exception raise when the name doesn't follow the AnDO specification
```
Comment (SylvainTakerkart) [inline ando/error.py:27]: same as above! check all the errors !

File: ando/tests/test_Rules.py [hunk 10]
```diff
@@ -1,3 +1,14 @@
+# File: test_Rules.py
+# Project: tests
+# File Created: Tuesday, 30th June 2020 10:50:05 am
+# Author: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Last Modified: Thursday, 2nd July 2020 1:35:37 pm
+# Modified By: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Copyright - 2020 MIT, Institue de neurosciences de la Timone
+
```
Comment (SylvainTakerkart) [inline ando/tests/test_Rules.py:10]: same

File: ando/tests/test_engine.py [hunk 11]
```diff
@@ -1,9 +1,18 @@
...
+# File: test_engine.py
+# Project: tests
+# File Created: Tuesday, 30th June 2020 10:50:05 am
+# Author: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Last Modified: Thursday, 2nd July 2020 1:33:13 pm
+# Modified By: garcia.j (Jeremy.garcia@univ-amu.fr)
+# -----
+# Copyright - 2020 MIT, Institue de neurosciences de la Timone
+
```
Comment (SylvainTakerkart) [inline ando/tests/test_engine.py:10]: same

**Review by SylvainTakerkart — CHANGES_REQUESTED** [review]

File: ando/_version.py [hunk 12]
```diff
@@ -1 +1 @@
-__version__ = '0.0.1'
+__version__ = '0.0.1b6'
```
[the remark this answers is not in the document]
Comment (Slowblitz) [reply ando/_version.py:1]: Ok so now it's 0.1 or 0.2 after this pull request ?

[the remark this answers is not in the document]
Comment (SylvainTakerkart) [reply ando/_version.py:1]: if that's ok with you, let's call it 0.2beta !

[the remark this answers is not in the document]
Comment (Slowblitz) [reply ando/_version.py:1]: Ok seems good to me

**Comment by SylvainTakerkart** [comment]
@Slowblitz can you reply to my comments if you fixed the correponsding thing?

File: ando/checker.py (hunk 1 above)
Comment (Slowblitz) [reply ando/checker.py:10]: Done

File: ando/engine.py (hunk 2 above)
Comment (Slowblitz) [reply ando/engine.py:10]: Done

File: ando/engine.py (hunk 3 above)
Comment (Slowblitz) [reply ando/engine.py]: Done

File: ando/engine.py (hunk 4 above)
Comment (Slowblitz) [reply ando/engine.py]: Done

File: ando/engine.py (hunk 5 above)
Comment (Slowblitz) [reply ando/engine.py]: Done

File: ando/engine.py (hunk 6 above)
Comment (Slowblitz) [reply ando/engine.py]: Done

File: ando/error.py (hunk 7 above)
Comment (Slowblitz) [reply ando/error.py:10]: Done

File: ando/error.py (hunk 8 above)
Comment (Slowblitz) [reply ando/error.py]: Done

File: ando/error.py (hunk 9 above)
Comment (Slowblitz) [reply ando/error.py]: Done

File: ando/tests/test_Rules.py (hunk 10 above)
Comment (Slowblitz) [reply ando/tests/test_Rules.py:10]: Done

**Comment by Slowblitz** [comment]
Do i comment and resolve ?

File: ando/engine.py [hunk 13]
```diff
@@ -234,26 +234,26 @@ def is_AnDO_verbose(directory):
...
     Raises:
-        ExperimentError: raised if it doesn't the experiment rules
-        SessionError: raised if it doesn't the session rules
-        SubjectError: raised if it doesn't the subject rules
-        SourceError: raised if it doesn't the source rules
-        RawDataError: raised if it doesn't the rawdata rules
-        DerivativeDataError: raised if it doesn't the derivatives rules
-        MetaDataError: raised if it doesn't the metadata rules
-        SourceNotFound: raised if it doesn't the source rules
+        ExperimentError: raised if it does not respect respect respect respect respect the experiment rules
```
Comment (SylvainTakerkart) [inline ando/engine.py:245]: this is a lot of respect ;)
and below also, there is some double respect!

**Review by SylvainTakerkart — CHANGES_REQUESTED** [review]

File: ando/engine.py (hunk 13 above)
Comment (Slowblitz) [reply ando/engine.py:245]: Hum I think there is a little bug I don't have it on my file 
image

**Review by SylvainTakerkart — APPROVED** [review]

**Pull request merged**